src/time_advance.cpp

```
@@ -512,14 +513,16 @@ imex_advance(PDE<P> &pde, matrix_list<P> &operator_matrices,
 
     pde.E_field.resize(dense_size);
 
-    // first_time = false;
+    first_time = false;
+    asgard::tools::timer.stop("update_system");
   }
 
   auto do_poisson_update = [&](fk::vector<P> const &f_in) {
     tools::timer.start("poisson_update");
     // Get 0th moment
-    fk::vector<P> mom0(dense_size);
-    fm::gemv(pde.moments[0].get_moment_matrix(), f_in, mom0);
+    fk::vector<P, mem_type::owner, resource::device> mom0(dense_size);
```

This is the kind of code that won't compile on the CPU. Before we made some of the recent changes, if CUDA was not enabled then `device` defaulted to the CPU, but that can be misleading. With the new changes, `device` cannot be used without CUDA.

Have a variable defined with pre-processor macros:
```
#ifdef ASGARD_USE_CUDA
constexpr resouce imex_resc = resouce::device;
#else 
constexpr resouce imex_resc = resouce::host;
#endif
```

The code may need a few more of these, but that's the way to go.